### PR TITLE
Convert front-end asset compiling engine down to a small NPM script

### PR DESCRIPTION
**Pull request by jediandrey**
Original Issue, this PR is going to fix: REPLACE WITH A LINK TO ISSUE ( publicly available )

Make sure these boxes are checked before asking for review of your pull request - thank you!

If there is a new feature or this is a bug fix - use 8.x-2.x branch. We'll tag for release if the bug is critical asap or tag for release next bug fix release until critical issue arrived.

## Steps for review

- [ ] Please provide steps for review here.
- [ ] Please provide steps for review here.
- [ ] Please provide steps for review here.
- [ ] Please provide steps for review here.

## General checks
- [ ] All coding styles are fulfilled and there are no any issues reported by CodeSniffer CI. <br/>CI code sniffer errors
- [ ] All tests are running and there are no failed tests reported by CI. <br/>Behat test results
- [ ] [Documentation](https://github.com/ymcatwincities/openy/tree/8.x-1.x/docs) has been updated according to PR changes.
- [ ] [Steps for review](https://github.com/ymcatwincities/openy/pull/94#issue-204580200) have been provided according to PR changes. <br/>Steps for review
- [ ] Make sure you've provided all necessary hook\_update\_N to [support upgrade path](https://github.com/ymcatwincities/openy/blob/8.x-1.x/docs/Development/Upgrade%20path.md).
- [ ] Make sure your git email is associated with account on drupal.org, otherwise you won't get commits there. <br/>drupal.org email
- [ ] If you would like to get credits on drupal.org, [check documentation](https://github.com/ymcatwincities/openy/blob/8.x-1.x/docs/Development/Contributing.md#drupalorg-credits).

Thank you for your contribution!

### CODE REVIEW

File: themes/openy_themes/openy_carnation/package.json [hunk 1]
```diff
@@ -9,35 +9,58 @@
...
     "stylelint-config-standard": "^18.3.0",
-    "time-grunt": "^2.0.0"
+    "time-grunt": "^2.0.0",
+    "webpack": "^4.41.2",
+    "webpack-cli": "^3.3.9",
+    "webpack-glob-entry": "^2.1.1",
+    "webpack-log": "^3.0.1"
+  },
+  "scripts": {
+    "build": "webpack --config webpack.build.js --debug --output-pathinfo",
```
Comment (podarok) [inline themes/openy_themes/openy_carnation/package.json:63]: let's add `--info-verbosity` to build process too

File: themes/openy_themes/openy_carnation/webpack.config.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+            {
+                test: /.js/,
+                enforce: 'pre',
+                exclude: [
+                    /node_modules/,
+                    /jshint-loader/,
+                    /webpack-entry.js/,
+//                    /.\/src\/js\/bootstrap.js/,
+//                    /jquery.matchHeight.js/,
+//                    path.resolve(__dirname, './src/js/bootstrap.js'),
```
Comment (podarok) [inline themes/openy_themes/openy_carnation/webpack.config.js]: can we get rid of commented code?

File: themes/openy_themes/openy_carnation/webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+                            sourceMap: true,
+                        }
+                    },
+                    {
+                        loader: 'sass-loader',
+                        options: {
+                            sourceMap: true,
+                        }
+                    },
+                    //'postcss-loader',
```
Comment (podarok) [inline themes/openy_themes/openy_carnation/webpack.config.js]: can we get rid of commented code?

**Review by podarok — CHANGES_REQUESTED** [review]
Let's ignore chmod when preparing this PR. We should keep 644, not 755 - see https://stackoverflow.com/a/1580644

**Comment by bjwilkins** [comment]
Build comment file: 
--------------

Check Open Y Installation Wizard at http://install.openy.cibox.tools/build1881/install.php

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy.cibox.tools:8080/job/PR_BUILDER_COMPOSER/1881/

**Comment by bjwilkins** [comment]
Acessibility Sniffer: front page checking WCAG2AA http://openy.cibox.tools/build1881/frontWCAG2AAhtmlcs.txt
Acessibility Sniffer: join page checking WCAG2AA http://openy.cibox.tools/build1881/joinWCAG2AAhtmlcs.txt
Acessibility Sniffer: locations page checking WCAG2AA http://openy.cibox.tools/build1881/locationsWCAG2AAhtmlcs.txt
Acessibility Sniffer: schedules page checking WCAG2AA http://openy.cibox.tools/build1881/schedulesWCAG2AAhtmlcs.txt
Acessibility Sniffer: blog page checking WCAG2AA http://openy.cibox.tools/build1881/blogWCAG2AAhtmlcs.txt

**Comment by anpolimus** [comment]
@podarok please do not merge this PR.
We need to test all carnation_refactoring together

**Pull request closed without merging**